testcases/crypto/ec_func.c

```diff
@@ -1196,15 +1200,29 @@ CK_RV run_GenerateSignVerifyECC(CK_SESSION_HANDLE session,
     if (parts > 0) {
         rc = funcs->C_SignFinal(session, signature, &signaturelen);
         if (rc != CKR_OK) {
-            testcase_error("C_SignFinal rc=%s", p11_get_ckr(rc));
-            goto testcase_cleanup;
+            if (rc == CKR_CURVE_NOT_SUPPORTED) {
```

Comment:
I guess there is no way to detect the invalid curve already a C_SignInit? That would probably be the better place, but if the ICA token can't find out if it supports the curve at C_SignInit, then this place is OK, too.

Reply:
Yep, there is no way. But as an aside: I saw that in ica_ec_key_new(), libica checks for supported curves via curve_supported(nid), which is just a switch over some known curves to be supported. There is another routine used in ecdsa_sign_sw: is_supported_openssl_curve(), which really tries to use a given curve in openssl. Maybe we can make a better check in ica_ec_key_new / ica_ec_key_init / ica_ec_key_generate so that an unsupported curve shows up when creating/initializing the key. But this would be a separate libica item.